data-spring-jdbc/src/main/java/io/micronaut/data/spring/jdbc/JdbcTemplateFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.spring.jdbc;
+
+import io.micronaut.context.annotation.EachBean;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.data.connection.jdbc.advice.DelegatingDataSource;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import javax.sql.DataSource;
+
+/**
+ * Factory for creating {@link JdbcTemplate} instances for each bean of type {@link DataSource}.
+ */
+@Factory
+class JdbcTemplateFactory {
```

Comment:
Please make final and internal

Reply:
resolved via https://github.com/micronaut-projects/micronaut-data/pull/2518/commits/3dbf01fedb59b86b2cd3d0e301aee2c7a39c15de